sdk/src/client/secret/mod.rs

```diff
@@ -195,6 +195,21 @@ impl Debug for SecretManager {
     }
 }
 
+impl fmt::Display for SecretManager {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            #[cfg(feature = "stronghold")]
+            Self::Stronghold(_) => write!(f, "Stronghold"),
+            #[cfg(feature = "ledger_nano")]
+            Self::LedgerNano(_) => write!(f, "LedgerNano"),
```

Comment:
Would this be interesting?
```suggestion
            Self::LedgerNano(l) => {
                if l.is_simulator {
                    write!(f, "LedgerNanoSimulator")
                } else {
                    write!(f, "LedgerNano")
                }
            }
```

Reply:
Probably! Done